### PR TITLE
Updates include: OFMCC-474 (assigned status update on reply), Fix to …

### PR DESCRIPTION
**Pull request by jgstorey**
Fixed:
OFMCC-754 QA bug (tooltip should not display when status is closed)
OFMCC-474 (assigned status occurs on reply).
Unread icon updated from open to closed envelope.
Small logic update to AssistanceRequestTable.vue requested by Viet.

### CODE REVIEW

File: frontend/src/components/messages/ReplyRequestDialog.vue [hunk 1]
```diff
@@ -97,28 +98,68 @@ export default {
...
         try {
           this.isLoading = true
-          const payload = {
-            assistanceRequestId: this.assistanceRequestId,
-            message: this.message,
-          }
-          await this.replyToAssistanceRequest(payload)
+          this.createReply(this.assistanceRequestId)
+          this.updateStatusToAssigned(this.assistanceRequestId)
+          this.updateStoredAssistanceRequest(this.assistanceRequestId)
```
Comment (weskubo-cgi) [inline frontend/src/components/messages/ReplyRequestDialog.vue:109]: You need to await each of these if you want them to execute in order.

File: frontend/src/components/messages/ReplyRequestDialog.vue [hunk 2]
```diff
@@ -97,28 +98,68 @@ export default {
...
           this.closeReplyRequestDialog()
           this.isLoading = false
         }
       }
     },
+
+    /**
+     * Create a reply for the assistance request.
+     */
+    async createReply(assistanceRequestId) {
```
Comment (weskubo-cgi) [inline frontend/src/components/messages/ReplyRequestDialog.vue:124]: I probably wouldn't bother creating a method for these three as they are all essentially one-liners (if you don't bother creating a variable for payload). But if you feel it helps understanding/error handling then it's fine.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/messages/ReplyRequestDialog.vue [hunk 3]
```diff
@@ -58,7 +59,7 @@ export default {
       type: String,
       required: true,
       default: '',
-    },
+    }
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/messages/ReplyRequestDialog.vue:62]: Mine was set to es5 as well. Made change manually.

File: frontend/src/components/messages/RequestConversations.vue [hunk 4]
```diff
@@ -170,6 +170,8 @@ export default {
      */
     async replySuccessEvent(isSuccess) {
       if (isSuccess) {
+        // assisantanceRequest status has been updated as part of reply, get the latest from store.
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/messages/RequestConversations.vue:173]: I was actually referencing the assistanceRequest object in comment. Regardless changed.

File: frontend/src/components/messages/ReplyRequestDialog.vue (hunk 1 above)
Comment (jgstorey) [reply frontend/src/components/messages/ReplyRequestDialog.vue:109]: Fixed.

File: frontend/src/components/messages/ReplyRequestDialog.vue (hunk 2 above)
Comment (jgstorey) [reply frontend/src/components/messages/ReplyRequestDialog.vue:124]: I did it for the error handling, i flipped and flopped on it but eventually thought the error handling was worth it given 3 api calls in one method.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**